### PR TITLE
Create .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by Shivansh2407**
BugZilla Link : https://bugzilla.mozilla.org/show_bug.cgi?id=1601820
Added a .pre-commit-config.yaml file

### CODE REVIEW

**Comment by Shivansh2407** [comment]
@armenzg please review the PR and let me know the changes that are needed to be done.

**Comment by armenzg** [comment]
@Shivansh2407 Please run this command locally (`yarn format:check`) and format the file correctly. You can use the flag `--write`

**Comment by armenzg** [comment]
Kyle graciously is taking the review for this!

**Comment by klahnakoski** [comment]
@Shivansh2407  I think this will require a few round of review.  

1. The first round will be updating the `requirements/dev.txt` to include a version of `pre-commit`:  Pick a version and hash that works for you.  Update the setup instructions (`docs/installation.md` )if needed.
2. Ensure the current tests run against your PR all pass:  I *think* you code is a no-op, so you should still be able to pass all the Travis checks.  I can answer your questions if this is not the case.
3. Then we can dig into each of the hooks you have added and ensure they are appropriate, and if they include what's in `runtests.sh`.

**Comment by Shivansh2407** [comment]
@klahnakoski can you please help me on how can I include version of pre-commit with hash?

**Comment by klahnakoski** [comment]
@Shivansh2407  Please look at `requirements/dev.txt`, which is where you will place the requirement for `pre-commit`.  You will see the hashes; follow that format.

You can find the hash by running `pip install -r requirements/dev.txt`, and noticing the complaint about missing hashes, which includes the correct hash. More information can be found here: https://github.com/klahnakoski/treeherder/blob/debug/tests/README.md (which  I am attempting to get merged into treeherder/master)

**Comment by Shivansh2407** [comment]
I have updated the requirements/dev.txt by running it on my local machine. Please check and let me know how to proceed further @klahnakoski .

File: requirements/dev.txt [hunk 1]
```diff
@@ -7,8 +7,15 @@ pytest==5.0.1 \
...
     --hash=sha256:9a23ada2e43cd989195db3c18710b5d7451134a0d48127ab64c1d2ad81700342 \
     --hash=sha256:17c53cd6bf4e7d69902aedf9a1d26c5d3b7369b54c5718744704f27b5a72f35d
-
-mock==3.0.5 --hash=sha256:d157e52d4e5b938c550f39eb2fd15610db062441a9c2747d3dbfa9298211d0f8 --hash=sha256:83657d894c90d5681d62155c82bda9c1187827525880eda8ff5df4ec813437c3
+Django==3.0.2 \
+    --hash=sha256:4f2c913303be4f874015993420bf0bd8fd2097a9c88e6b49c6a92f9bdd3fb13a
+sqlparse==0.3.0 \
+    --hash=sha256:40afe6b8d4b1117e7dff5504d7a8ce07d9a1b15aeeade8a2d10f130a834f8177
+asgiref==3.2.3 \
+    --hash=sha256:ea448f92fc35a0ef4b1508f53a04c4670255a3f33d22a81c8fc9c872036adbe5
```
Comment (klahnakoski) [inline requirements/dev.txt]: These three are already in `common.txt`.  Remove these, and update the other file, if needed.

File: requirements/dev.txt [hunk 2]
```diff
@@ -162,8 +171,8 @@ zope.deprecation==4.4.0 \
     --hash=sha256:f1480b74995958b24ce37b0ef04d3663d2683e5d6debc96726eff18acf4ea113 \
     --hash=sha256:0d453338f04bacf91bbfba545d8bcdf529aa829e67b705eac8c1a7fdce66e2df
 zope.proxy==4.3.2 \
-    --hash=sha256:ab6d6975d9c51c13cac828ff03168de21fb562b0664c59bcdc4a4b10f39a5b17
+    --hash=sha256:af10cb772391772463f65a58348e2de5ecc06693c16d2078be276dc068bcbb54
```
Comment (klahnakoski) [inline requirements/dev.txt]: Why did this hash change?

Comment (Shivansh2407) [reply requirements/dev.txt]: I think it was updated to a newer version or something after installation on my local machine...I have removed the code from dev.txt.
Django==3.0.2 \
mock==3.0.5 --hash=sha256:d157e52d4e5b938c550f39eb2fd15610db062441a9c2747d3dbfa9298211d0f8 --hash=sha256:83657d894c90d5681d62155c82bda9c1187827525880eda8ff5df4ec813437c3	    --hash=sha256:4f2c913303be4f874015993420bf0bd8fd2097a9c88e6b49c6a92f9bdd3fb13a
sqlparse==0.3.0 \
    --hash=sha256:40afe6b8d4b1117e7dff5504d7a8ce07d9a1b15aeeade8a2d10f130a834f8177
asgiref==3.2.3 \
    --hash=sha256:ea448f92fc35a0ef4b1508f53a04c4670255a3f33d22a81c8fc9c872036adbe5


Do I need to update these hash in common.txt??

Comment (Shivansh2407) [reply requirements/dev.txt]: Any update @klahnakoski ?

Comment (klahnakoski) [reply requirements/dev.txt]: Do not replace the hash.  Use both hashes.  I see your new hash here

https://pypi.org/project/zope.proxy/4.3.2/#files

Comment (klahnakoski) [reply requirements/dev.txt]: @Shivansh2407   ...also, please @klahnakoski, otherwise I may not see your comments

**Comment by armenzg** [comment]
A smaller version of it landed few weeks back.

**Pull request closed without merging**